docs/enhancements/agent-late-binding.md

```diff
@@ -53,28 +53,29 @@ There are two main (non-breaking) changes to the CRDs:
 1. ClusterRef in the Agent Spec will no longer be mandatory.
 
-A Discovery ISO with no associated cluster means that hosts booted with that ISO will have agents that are not bound to any cluster.  Thus, Agent CRs can be created with no cluster binding.  It will still be possible to retain the previous behavior by providing a ClusterRef.  The multitenancy model will remain unchanged, with Agents being created in the namespace of the InfraEnv they booted from.
+A Discovery ISO with no associated cluster means that hosts booted with that ISO will have agents that are not bound to any cluster.  Thus, Agent CRs can be created with no cluster binding.  It will still be possible to retain the previous behavior by providing a ClusterRef.  The tenancy model will remain unchanged, with Agents being created in the namespace of the InfraEnv they booted from.
 
 The Agent's ClusterRef may now be deleted or changed dynamically, unless the Agent is part of an ongoing cluster installation, in which case the installation must first be cancelled (not currently possible via the k8s API).  If the host had started/completed its installation, a new Agent Condition would indicate that the user needs to boot the host into the Discovery ISO once
 again (as mentioned in the non-goals, this can be done automatically but is not in the scope of this enhancement).
 
-The REST API will require bigger changes:
-1. Move images to be a standalone resource, rather than part of the cluster resource (currently /clusters/{cluster_id}/downloads/image).  This removes the dependency of an image on a specific cluster, removes the current limitation of one image per cluster, and aligns with InfraEnv CRD.  The image resource will have an additional optional cluster-id property.
-1. Move hosts to be a standalone resource, rather than under of the cluster resource (currently /clusters/{cluster_id}/hosts/{host_id}), and add add an optional cluster-id property.
+The REST API will require breaking changes, so we will create a v2 API.  At the same time we will align with the k8s API (change host to agent, and image to infra-env).
+1. Move images (infra-envs) to be a standalone resource, rather than part of the cluster resource (currently `/clusters/{cluster_id}/downloads/image`).  This removes the dependency of an image on a specific cluster, removes the current limitation of one image per cluster, and aligns with InfraEnv CRD.
+1. An agent is now always referenced using a combination of the infra-env ID and host ID, and the REST API path will be `/infra-envs/{infra_env_id}/hosts/{host_id}` rather than `/clusters/{cluster_id}/hosts/{host_id}`.  This means that a server that is booted from the ISO of infra-env1 and then from the ISO of infra-env2 will generate two Agents (the first's state should be `disconnected`).  The Agent will have an optional cluster-id property (foreign key in the DB) that signifies which cluster the Agent is bound to, if any.
```

Comment:
didn't we want to change it to `/infra-envs/{infra_env_id}/agents/{agent_id}` ?

Reply:
Yes, I was concentrating on the cluster->infra-env change